rewrite-java/src/main/java/org/openrewrite/java/RenameVariable.java

```diff
@@ -152,18 +152,53 @@ private boolean isVariableName(Object value, J.Identifier ident) {
          * FieldAccess targets the variable if its target is an Identifier and either
          * its target FieldType equals variable.Name.FieldType
          * or its target Type equals variable.Name.FieldType.Owner
+         * or if FieldAccess targets a TypCast and either
+         * its type equals variable.Name.FieldType
+         * or its type equals variable.Name.FieldType.Owner.
+         * In case the FieldAccess targets another FieldAccess, the target is followed
+         * until it is either an Identifier or a TypeCast.
          */
         private boolean fieldAccessTargetsVariable(J.FieldAccess fieldAccess) {
-            if (renameVariable.getName().getFieldType() != null && fieldAccess.getTarget() instanceof J.Identifier) {
-                J.Identifier fieldAccessTarget = (J.Identifier) fieldAccess.getTarget();
+            if (renameVariable.getName().getFieldType() != null) {
+                Expression target = getTarget(fieldAccess);
+                JavaType targetType = resolveType(target.getType());
                 JavaType.Variable variableNameFieldType = renameVariable.getName().getFieldType();
-                JavaType fieldAccessTargetType = fieldAccessTarget.getType() instanceof JavaType.Parameterized ? ((JavaType.Parameterized) fieldAccessTarget.getType()).getType() : fieldAccessTarget.getType();
-                return variableNameFieldType.equals(fieldAccessTarget.getFieldType()) ||
-                       (fieldAccessTargetType != null && fieldAccessTargetType.equals(variableNameFieldType.getOwner()));
+                if (TypeUtils.isOfType(variableNameFieldType.getOwner(), targetType)) {
+                    return true;
```

Comment:
This early return might need your attention @renegrob ; I've extracted this from seeing it in both if/else/if blocks below in your earlier commit.

Reply:
Looks good to me. Thank you so much!